Security documentation fixup

thoh-ot opened the pull request:
Various fixes to the security documentation as discussed on the Security Subcommittee meetings.

thoh-ot committed:
doc: security: Fix section reference

Replace reference to Chapter 4 with Secure Design section.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot committed:
doc: security: Unify citations

Use similar style for citations.
Add online references where applicable.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

dbkinder commented on doc/security/secure-coding.rst:
```diff
@@ -262,7 +258,7 @@ project itself, it may be necessary to increase this embargo time.
 
 The list of issues shall be reviewed at least once a month by the
-security committee on the Zephyr Project.  This review should focus on
+Zehyr Security Subcommittee.  This review should focus on
```
```suggestion
Zephyr Security Subcommittee.  This review should focus on
```

dbkinder commented on doc/security/security-overview.rst:
```diff
@@ -105,8 +105,8 @@ The three major security measures currently implemented are:
    blocks such as network stacks increases the overall quality level
-   and guarantees stable APIs. Static code analyses are planned for
-   the near future.
+   and guarantees stable APIs. Static code analyses is provided by
```
Since ``analyses`` is plural...
```suggestion
   and guarantees stable APIs. Static code analyses are provided by
```

dbkinder commented on doc/security/security-overview.rst:
```diff
@@ -380,16 +380,17 @@ The quality assurance part encompasses the following criteria:
-   process documentation. Waivers shall be documented centrally and
+   waive potential false-positives before each release.
+   Waivers shall be documented centrally and
    in form of a comment inside the source code itself. The
```
change to "in the form of"

dbkinder commented on doc/security/security-overview.rst:
```diff
@@ -551,27 +552,27 @@ following is the current proposal:
    of the Security Group; members of this Security Group are:
 
-   -  members of the Security Subcommittee
+   -  members of the Zephyr Security Subcommittee
```
This list should just describe who the members are, so rewrite as:
```
   -  members of the Zephyr Security Subcommittee
   -  others, as proposed and ratified by the Zephyr Security Subcommittee
   -  the reporter

-  Members of this Security Group have the authority to add or remove other
   users for individual issues.   
```
Note that last bullet is outdented to belong to the outer bullet list (not part of the list
defining who the members are).

dbkinder reviewed (changes requested):
Also, do you have a resolution for the few TBDs still in this doc?

thoh-ot committed:
doc: security: Unify name of security subcommittee

Use the term 'Zephyr Security Subcommittee' where applicable.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot committed:
doc: security: Correct statement about Coverity

Remove statement that static code analysis is planned for the future.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot committed:
doc: security: Reword unintelligible sentences

The sentence "To process process documentation." does not make
any sense at all.

Add missing "the" to the sentence "in form of".

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot committed:
doc: security: Update Security Group definition

Rewrite who the members of the Security Group are
and move the 'ability' of the members to an outer
bullet point.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot commented:
@dbkinder I do not have any resolution for the TBD's in the documentation, I will raise the question on tomorrows Zephyr Security Subcommittee meeting.

thoh-ot committed:
doc: security: Correct indentation

Use similar indentation as the other bullet points above.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

thoh-ot committed:
doc: security: Update embargo period

The embargo period is 60 days.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>

dbkinder commented on doc/security/security-overview.rst:
```diff
@@ -380,17 +380,18 @@ The quality assurance part encompasses the following criteria:
-   security reviews as laid out in Chapter 4. Please refer to the
-   `development model documentation`_ on the Zephyr project Wiki.
+   security reviews as laid out in the `Secure Design`_ section.
+   Please refer to the `development model documentation`_ on the
```
The dev model docs are now part of the technical docs, so let's change this to:
```
Refer to the :ref:`development_model` documentation for more information.
```

thoh-ot replied:
@dbkinder Pushed an update for this

dbkinder commented on doc/security/security-overview.rst:
```diff
@@ -778,13 +778,4 @@ include [GHS10]_
-.. _`RFC2119`: https://www.ietf.org/rfc/rfc2119.txt
-.. _`Application Thread Modeling`: https://www.owasp.org/index.php/Application_Threat_Modeling
-.. _`STRIDE`: https://msdn.microsoft.com/en-us/library/ee823878%28v=cs.20%29.aspx
 .. _`development model documentation`: https://github.com/zephyrproject-rtos/zephyr/wiki/Development-Model
```
... and delete this line

dbkinder reviewed (changes requested):
dev model docs moved from the wiki to the technical docs...

ruuddw reviewed (approved):
Thanks for reviewing and improving the security docs: consistent use of 'Zephyr Security WG', references, TBD fixes, etc.

thoh-ot committed:
doc: security: Change link to development model.

The development model docs are now part of the technical docs.

Signed-off-by: Thomas Ebert Hansen <thoh@oticon.com>
